operator/v1/types_ingress.go

```diff
@@ -376,6 +386,21 @@ type LoadBalancerStrategy struct {
 	// +required
 	Scope LoadBalancerScope `json:"scope"`
 
+	// allowedSourceRanges specifies an allowlist of IP address ranges to which
+	// access to the load balancer should be restricted.  Each range must be
+	// specified using CIDR notation (e.g. "10.0.0.0/8" or "fd00::/8"). If no range is
+	// specified, "0.0.0.0/0" for IPv4 and "::/0" for IPv6 are used by default,
+	// which allows all source addresses.
+	//
+	// To facilitate migration from earlier versions of OpenShift that did not have the
+	// allowedSourceRanges field, the service.beta.kubernetes.io/load-balancer-source-ranges
+	// annotation on the load balancer service takes precedence on OpenShift v4.12
```

Comment:
it takes precedence, it's not used in combination, so if I specify value/A here and value/B on the annotation, I get value/B and this gets ignored?

Can we write a control loop that actually stomps this when that is happening so that a user knows the state of the cluster?  Otherwise a client cannot rely on this API and must interpret two different APIs to determine his configuration.

Reply:
> Can we write a control loop that actually stomps this when that is happening so that a user knows the state of the cluster? Otherwise a client cannot rely on this API and must interpret two different APIs to determine his configuration.

`allowedSourceRanges` appears in `spec` and in `status`.  Would it be appropriate to reflect the annotation (if it's set) to `status.allowedSourceRanges` when `spec.allowedSourceRanges` is empty?

Reply:
I've updated the [EP](https://github.com/openshift/enhancements/pull/1177/files#diff-2182da078aabca9029bf489d67b4e726c7fc72fcf4878cae69a6a1d08f3ff249R138) to state `status.allowedSourceRanges` will reflect the current effective value by looking at the annotation and the `spec.loadBalancerSourceRanges` field.